src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/UriValidator.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Checks if a URI is structurally valid
+    /// </summary>
+    public class UriValidator : IOptionSettingItemValidator
+    {
+        public string ValidationFailedMessage { get; set; } = "{{URI}} is not a valid URI.";
+
+        public Task<ValidationResult> Validate(object input, Recommendation recommendation)
+        {
+            var uri = input?.ToString() ?? string.Empty;
+
+            if (string.IsNullOrEmpty(uri))
+            {
+                return ValidationResult.FailedAsync("Invalid URI. The URI cannot be null or empty");
```

Comment:
Is true of all future `Uri` options? 

If not and we may conceivably add an option Uri option in the future, I'd rather use a `Required` validator here to enforce that a value is provided.

Reply:
I cannot foresee such a use case right now. But even if we encounter it in the future than it should be easy to remove the
`string.IsNullOrEmpty(uri)`  check. Also, this removal would not be a breaking change as we would be making the validator more permissive.

Reply:
What if a user is using a custom recipe with a Uri option? Could they modify the source of the validators?

In general I would prefer validators as self-contained and composable as possible.